terminus-ui/confirmation/src/confirmation.directive.ts

```diff
@@ -162,15 +203,51 @@ export class TsConfirmationDirective implements OnDestroy, OnInit {
    * Create and attach the confirmation overlay
    */
   private createOverlay(): void {
+    let modalPosOriginX: HorizontalConnectionPos = 'center';
+    let modalPosOriginY: VerticalConnectionPos = 'bottom';
+    let modalPosOverlayX: HorizontalConnectionPos = 'center';
+    let modalPosOverlayY: VerticalConnectionPos = 'top';
+    let positionClass = 'ts-confirmation-overlay__panel-below';
+    switch (this.modalPosition) {
+      case ('above'):
+        modalPosOriginY = 'top';
+        modalPosOverlayY = 'bottom';
+        positionClass = 'ts-confirmation-overlay__panel-above';
+        break;
+      case ('below'):
+        modalPosOriginY = 'bottom';
+        modalPosOverlayY = 'top';
+        positionClass = 'ts-confirmation-overlay__panel-below';
+        break;
+      case ('before'):
+        modalPosOriginX = 'start';
+        modalPosOriginY = 'center';
+        modalPosOverlayX = 'end';
+        modalPosOverlayY = 'center';
+        positionClass = 'ts-confirmation-overlay__panel-before';
+        break;
+      case ('after'):
+        modalPosOriginX = 'end';
+        modalPosOriginY = 'center';
+        modalPosOverlayX = 'start';
+        modalPosOverlayY = 'center';
+        positionClass = 'ts-confirmation-overlay__panel-after';
+        break;
+      default:
+        modalPosOriginY = 'bottom';
+        modalPosOverlayY = 'top';
+        positionClass = 'ts-confirmation-overlay__panel-below';
+        break;
+    }
```

Comment:
I'd love to see our config generation happen in another function just to keep concerns separated and methods smaller. Two possible options:

a) The `modalPosition` input setter calls a function each time it is set:

```typescript
set modalPosition(v) {
  this._modalPosition = v;
  this.overlayConfig = this.generateOverlayConfig(v);
}
```

b) The `createOverlay` method calls it directly:

```typescript
createOverlay() {
  const overlayConfig = this.generateOverlayConfig(this.modalPosition);
  ...
```

(in both examples I pass the position to the generate function which keeps our method 'pure')

Reply:
I went with option B